AK/Statistics.h

```diff
@@ -27,10 +27,23 @@ class Statistics {
     }
 
     T const sum() const { return m_sum; }
-    float average() const { return (float)sum() / size(); }
+
+    // FIXME: Unclear Wording, average can mean a lot of different things
+    // Median, Arithmetic Mean (which this is), Geometric Mean, Harmonic Mean etc
+    float average() const
+    {
+        // Let's assume the average of an empty dataset is 0
+        if (size() == 0)
+            return 0;
+
+        return (float)sum() / size();
+    }
 
     T const min() const
     {
+        // Lets Rather fail than read over the end of a collection
+        VERIFY(size() != 0);
```

Comment:
most of numerals are linearly ordered; maybe we could define a `minimal` one (i.e. lowest than any other numeral) and use it just like zero is used in the `average`? (there it is just neutral element of sum, which could be defined as `zero`)

pretty much the same for maximum below? 🤔

Reply:
I've thought about using something like INT_MAX for min and INT_MIN for max (since those would be also useful starting values for the algorithm instead of using the first values). But Since we are generic, I'd need to find a way to find the upper and lower bounds of a template type which I wasn't able to find anywhere in the serenity codebase (only searched for like 5 mins though) so i just put a verify there for now.
If you know a good way to get the bounds of a templated types, that would be useful to fix this.

Reply:
it’s pretty generic so it has potential to be used all over the place—it’ll be better to check with maintainers how they see that…

Reply:
Yeah, I think I leave it like this for now, crashing at a verify with a comment is better than crashing from a vector out of bounds or returning something unexpected.